Include/internal/pycore_frame.h

```diff
@@ -194,7 +194,8 @@ _PyFrame_LocalsToFast(_PyInterpreterFrame *frame, int clear);
 static inline bool
 _PyThreadState_HasStackSpace(PyThreadState *tstate, int size)
 {
-    return tstate->datastack_top + size < tstate->datastack_limit;
+    return tstate->datastack_top != NULL &&
+    tstate->datastack_top + size < tstate->datastack_limit;
```

Comment:
`tstate->datastack_top + size` may be an undefined behavior if it is past the array end.

`size < tstate->datastack_limit - tstate->datastack_top` does not have an UB in that case.

I do not know whether this actually happens.

Reply:
To be precise, `tstate->datastack_limit - tstate->datastack_top` needs both pointers to point to entries in the same array to be defined (I think).